service/storage_proxy.cc

```diff
@@ -4136,6 +4136,32 @@ storage_proxy::query_partition_key_range_concurrent(storage_proxy::clock_type::t
                 break;
             }
 
+            // Checks if 2 range bounds are contiguous, permits false negatives
```

Comment:
At this stage, it's impossible to understand what "false negatives" means in this context, and why they are ok. You have a longer comment below explaining exactly which sense of "contiguous" you need here, and why it's fine that we don't check for two adjacent **inclusive** bounds. I would move this comment here - it will make it clearer that "are_contiguous" is just a hack that we need in this place, and not a general function that can be used in other places.

But even better - I wouldn't make this a function (lambda at all). This is just *one line of code* (a boolean expression) with a long comment in front of it, called in one place. You can move this one line of code - and the comment - to where you call it. When something is a hacky boolean condition only intended to work in one particular place in the code, I think it's better not to make it into a useful-looking function.